plugins/powerd/fu-plugin-powerd.c

```diff
@@ -34,40 +32,28 @@ gboolean
 fu_plugin_startup (FuPlugin *plugin, GError **error)
 {
 	FuPluginData *data = fu_plugin_get_data (plugin);
-	g_autoptr(GDBusConnection) connection = NULL;
 	g_autofree gchar *name_owner = NULL;
 
-	/* establishing DBus connection and proxy for method call to powerd */
-	connection = g_bus_get_sync (G_BUS_TYPE_SYSTEM, NULL, error);
-
-	if (connection == NULL) {
-		g_prefix_error (error,
-				"Connection with system was not established: ");
-		return FALSE;
-	}
-
-	data->proxy = g_dbus_proxy_new_sync (connection,
-					     G_DBUS_PROXY_FLAGS_NONE,
-					     NULL,
-					     "org.chromium.PowerManager",
-					     "/org/chromium/PowerManager",
-					     "org.chromium.PowerManager",
-					     NULL,
-					     error);
+	/* establish proxy for method call to powerd */
+	data->proxy = g_dbus_proxy_new_for_bus_sync (G_BUS_TYPE_SYSTEM,
```

Comment:
@g2pop and @binaryNewts, was there a reason to use the `g_dbus_proxy_new` instead of `g_dbus_proxy_new_for_bus_sync` ? 

I admit I'm not 100% familiar with the difference.

Reply:
The only use of getting the connection separately is if you want to do low level filtering, or watch for the closed signal -- but for a system bus that's realistically never going to happen. From a speed point of view it's probably unmeasurable, as we'll just be getting a copy of the singleton instance -- it's just less LoC.

Reply:
Right, like @hughsie implied, g_dbus_proxy_new_for_bus_sync () basically combines the previous two functions, g_bus_get_sync () (connecting to a bus) and g_dbus_proxy_new_sync () (making a proxy for it). It was easier to see the steps and know what was happening when they were separated (like in https://github.com/fwupd/fwupd/blob/HEAD/libfwupdplugin/fu-common-linux.c#L34), but there's no real difference except for what @hughsie mentioned.